export_snowflake/__init__.py

```diff
@@ -522,26 +528,126 @@ def flush_records(stream: str,
     db_sync.delete_from_stage(stream, s3_key)
 
 
+def transfer_data_in_parallel(s3_csv_lists, config, o, file_format_type):
+    parallel_transfer_start_time = time.time()
+    
+    # make the decision of how many processes to use later
+    pool = multiprocessing.Pool(4, initialize_s3_client_db_sync(config, o, file_format_type))
+    pool.map(download_file, s3_csv_lists)
+    pool.close()
+    pool.join()
+    
+    parallel_transfer_end_time = time.time()
+    stream = config["stream"]
+    LOGGER.info(f"Start time for transfering {stream} as {parallel_transfer_start_time} and end time as {parallel_transfer_end_time}")
+
+
+def download_file(job):
+    bucket, key, filename, stream = job
+
+    s3_client.download_file(bucket, key, filename)
+
+    # upload to snowflake
+    file_path = filename.replace("\\", "/")
+
+    s3_key = db_sync.put_to_stage(file_path, stream, temp_dir=None)
+
+    # remove from local after uploading to snowflake
+    os.remove(file_path)
+
+
+def export_to_snowflake(config, o, file_format_type): 
+    # retrieve the csv.gz lists and the schema file from the s3 bucket
+    s3_client = boto3.client('s3')
+
+    bucket = config["bucket"]
+    prefix = config["prefix"]
+    response = s3_client.list_objects_v2(Bucket=bucket, Prefix=prefix)
+
+    s3_csv_lists = []
+    schema_file = None
+    for obj in response['Contents']:
+        if obj['Key'].endswith('.csv.gz'):
+            s3_csv_lists.append((bucket, obj['Key'], obj['Key'].split("/")[-1], config["stream"]))
+        # for both python and spark parquet_to_csv steps, we always have a schema.json with the schema details
+        if obj['Key'].endswith('.json'):
+            schema_file = obj['Key']
+
+    if schema_file is None:
+        raise Exception("Schema file not found in the s3 bucket")
+    if len(s3_csv_lists) == 0:
+        raise Exception("No csv files found in the s3 bucket")
+
+    local_schema_file = 'local_schema.json'
+    s3_client.download_file(bucket, schema_file, local_schema_file)
+
+    f = open(local_schema_file)
+    schema = json.load(f)
+
+    # generate the schema object for DbSync
+    stream = config["stream"]
+    o = {
+        "stream": stream,
+        "schema": {"properties": schema['properties']},
+        "key_properties": config["key_columns"]
+    }
+
+    # check if the schema(Snowflake table) exists and create it if not
+    # sync_table will check the schema with the table in Snowflake, if there's a miss matching in columns, raise an error
+    db_sync = DbSync(config, o, None, file_format_type)
+    db_sync.create_schema_if_not_exists()
+    db_sync.sync_table()
+
+    # the csv files in s3 are compressed, changde the parquet_to_csv to write into gzip format, also remove the headers
```

Comment:
Is this change only for export snowflake? or it impacts all exports?

Reply:
Will only apply to snowflake export, all the changes relates to parquet_to_csv are flagged specifically for snowflake